hazelcast/src/test/java/com/hazelcast/map/AbstractMapIteratorTest.java

```diff
@@ -0,0 +1,256 @@
+        String value = randomString();
+        map.put(key, value);
+
+        Iterator<Map.Entry<Object, Object>> iterator = getIterator(map);
+        iterator.next();
+        iterator.remove();
+        assertEquals(0, map.size());
+    }
+
+    @Test
+    public void test_remove_withMultiplePartitions() {
+        IMap<Object, Object> map = instance.getMap(randomMapName());
+
+        String key = generateKeyForPartition(instance, 0);
+        String value = randomString();
+        map.put(key, value);
+
+        String key2 = generateKeyForPartition(instance, 1);
+        String value2 = randomString();
+        map.put(key2, value2);
+
+        Iterator<Map.Entry<Object, Object>> iterator = getIterator(map);
+        iterator.next();
+        iterator.remove();
+        assertEquals(1, map.size());
+        iterator.next();
+        iterator.remove();
+        assertEquals(0, map.size());
+    }
+
+    @Test
+    public void test_HasNext_Returns_False_On_EmptyMap() {
+        IMap<Object, Object> map = instance.getMap(randomMapName());
+        Iterator<Map.Entry<Object, Object>> iterator = getIterator(map);
+        assertFalse(iterator.hasNext());
+    }
+
+    @Test
+    public void test_HasNext_Returns_True_On_NonEmptyMap() {
+        IMap<Object, Object> map = instance.getMap(randomMapName());
+
+        String key = generateKeyForPartition(instance, 1);
+        String value = randomString();
+        map.put(key, value);
+
+        Iterator<Map.Entry<Object, Object>> iterator = getIterator(map);
+        assertTrue(iterator.hasNext());
+    }
+
+    @Test
+    public void test_Next_Returns_Value_On_NonEmptyMap() {
+        IMap<String, String> map = instance.getMap(randomMapName());
+
+        String key = generateKeyForPartition(instance, 1);
+        String value = randomString();
+        map.put(key, value);
+
+        Iterator<Map.Entry<String, String>> iterator = getIterator(map);
+        Map.Entry<String, String> entry = iterator.next();
+        assertEquals(value, entry.getValue());
+    }
+
+    @Test
+    public void test_Next_Returns_Value_On_NonEmptyMap_and_HasNext_Returns_False_when_Item_Consumed() {
+        IMap<String, String> map = instance.getMap(randomMapName());
+
+        String key = generateKeyForPartition(instance, 1);
+        String value = randomString();
+        map.put(key, value);
+
+        Iterator<Map.Entry<String, String>> iterator = getIterator(map);
+        Map.Entry<String, String> entry = iterator.next();
+        assertEquals(value, entry.getValue());
+        boolean hasNext = iterator.hasNext();
+        assertFalse(hasNext);
+    }
+
+    @Test
+    public void test_Next_Returns_Values_When_FetchSizeExceeds_On_NonEmptyMap() {
+        IMap<String, String> map = instance.getMap(randomMapName());
+
+        String value = randomString();
+        for (int i = 0; i < 100; i++) {
+            String key = generateKeyForPartition(instance, 1);
+            map.put(key, value);
+        }
```

Comment:
can be replaced with `putValuesToPartition(instance, map, randomString(), 1, 100)`